frontend/src/components/modules/Tournaments/OngoingTournament/TeamRoundRobin/TeamRoundRobinUpcomingView.tsx

```diff
@@ -16,24 +16,101 @@ import Paper from "@mui/material/Paper";
 import { useTranslation } from "react-i18next";
 import { Grid, Link } from "@mui/material";
 import CopyToClipboardButton from "../../OngoingTournament/CopyToClipboardButton";
+import api from "api/axios";
+import useToast from "hooks/useToast";
 
 const TeamRoundRobinUpcomingView: React.FC = () => {
   const { t } = useTranslation();
   const navigate = useNavigate();
   const { userId } = useAuth();
   const tournament = useTournament();
+  const showToast = useToast();
 
-  const userAlreadySigned = tournament.players.some(
-    (player) => player.id === userId
+  const userTeam = tournament.teams?.find((team) =>
+    team.players.some((player) => player.id === userId)
   );
+
+  const maxTeamsReached =
+    (tournament.teams?.length ?? 0) >= (tournament.numberOfTeams ?? Infinity);
+
+  const isUserTheCreator = tournament.creator.id === userId;
   const maxPlayers = tournament.maxPlayers;
   const signedPlayers = tournament.players.length;
   const tournamentFull = maxPlayers <= signedPlayers;
 
+  const handleJoinTeam = async (teamId: string): Promise<void> => {
+    if (
+      tournament?.id === undefined ||
+      tournament.id === "" ||
+      teamId === undefined ||
+      teamId === "" ||
+      userId === undefined ||
+      userId === ""
+    ) {
+      showToast(t("messages.invalid_team_join_data"), "error");
+      return;
+    }
+
+    try {
+      await api.tournaments.joinTeam(tournament.id, teamId, userId);
+      showToast(t("messages.team_join_success"), "success");
+      window.location.reload();
+    } catch (error) {
+      showToast(t("messages.team_join_error"), "error");
+    }
+  };
+
+  const handleLeaveTeam = async (teamId: string): Promise<void> => {
+    if (
+      tournament?.id === undefined ||
+      tournament.id === "" ||
+      teamId === undefined ||
+      teamId === "" ||
+      userId === undefined ||
+      userId === ""
+    ) {
+      showToast(t("messages.invalid_team_leave_data"), "error");
+      return;
+    }
+
+    try {
+      await api.tournaments.leaveTeam(tournament.id, teamId, userId);
+      showToast(t("messages.team_leave_success"), "success");
+      window.location.reload();
+    } catch (error) {
+      showToast(t("messages.team_leave_error"), "error");
+    }
+  };
+
+  const handleKickPlayer = async (
+    teamId: string,
+    playerId: string
+  ): Promise<void> => {
+    try {
+      await api.tournaments.kickPlayerFromTeam(tournament.id, teamId, playerId);
+      showToast(t("messages.player_kicked_success"), "success");
+      window.location.reload();
+    } catch (error) {
+      showToast(t("messages.player_kick_error"), "error");
+    }
+  };
+
+  const handleDeleteTeam = async (teamId: string): Promise<void> => {
+    try {
+      console.log("hi");
```

Comment:
accidental?

Reply:
Oh... Yeah, it shouldn't be there. I'll remove it next time